src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/MetricsEventSource.cs

```diff
@@ -115,40 +115,43 @@ public void CollectionStop(string sessionId, DateTime intervalStartTime, DateTim
             WriteEvent(3, sessionId, intervalStartTime, intervalEndTime);
         }
 
-        [Event(4, Keywords = Keywords.TimeSeriesValues, Version = 1)]
+        [Event(4, Keywords = Keywords.TimeSeriesValues, Version = 2)]
 #if !NET8_0_OR_GREATER
         [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2026:RequiresUnreferencedCode",
                                       Justification = "This calls WriteEvent with all primitive arguments which is safe. Primitives are always serialized properly.")]
 #endif
-        public void CounterRateValuePublished(string sessionId, string meterName, string? meterVersion, string instrumentName, string? unit, string tags, string rate, string value)
+        public void CounterRateValuePublished(string sessionId, string meterName, string? meterVersion, string instrumentName, string? unit, string tags, string rate, string value,
+                                    string instrumentFormattedTags, string instrumentHash, string meterFormattedTags, string meterHash, string formattedScopeHash)
         {
-            WriteEvent(4, sessionId, meterName, meterVersion ?? "", instrumentName, unit ?? "", tags, rate, value);
+            WriteEvent(4, sessionId, meterName, meterVersion ?? "", instrumentName, unit ?? "", tags, rate, value, instrumentFormattedTags, instrumentHash, meterFormattedTags, meterHash, formattedScopeHash);
```

Comment:
We're sending this extra information now on every published value, even though (it looks like) the values don't change for a given instrument? Is that adding a lot of overhead?

Reply:
While it is more data than before, I'm not anticipating it to be problematic. The MetricsEventSource has a default limit at 1000 time series being sent at most once per second. Adding these tags and hashes may mean an extra ~100KB/sec is flowing but that is still a couple orders of magnitude what we expect to be stressful amount of data. Consumers can also control the number of time series or the frequency of updates to further reduce the volume. If it did prove to be an issue we could work on some feature in the future to more carefully curate the data provided.

Reply:
@CodeBlanch - I just noticed your comment https://github.com/dotnet/runtime/issues/93767#issuecomment-2137869987 which also refers to this. I'd be fine having these metric data events include an instrument ID and not repeat the other data but two concerns:
1. Tarek has a limited amount of time with the deadline coming very soon. You guys would either need to agree on this fast, or we convince Tarek's managers that its worth making the change after the deadline.
2. You need to add a unique ID per instrument. The Instrument hashcode is close to that, but probably not a strong enough uniqueness guarantee that you'd want to rely on it as a join key. It doesn't take a ridiculous number of Instruments to get an occasional collision in 32 bit numbers. You could get a stronger ID by assigning a unique integer to each InstrumentState when it is first built.